helix-core/src/textobject.rs

```diff
@@ -112,6 +112,35 @@ pub fn textobject_word(
     }
 }
 
+// `textobject` and `count` are unused for now, this textobject behaves the same
+// whether it is inside or around.
+pub fn textobject_whitespace(
+    slice: RopeSlice,
+    range: Range,
+    _textobject: TextObject,
+    _count: usize,
+) -> Range {
+    let pos = range.cursor(slice);
+
+    // find the number of consecutive whitespace chars in front of the cursor
+    let whitespace_start = pos
+        - slice
+            .chars_at(slice.len_chars())
+            .reversed()
+            .skip(slice.len_chars() - pos - 1)
+            .take_while(|c| char_is_whitespace(*c) || *c == '\r' || *c == '\n')
```

Comment:
```suggestion
            .take_while(|*c| c.is_whitespace())
```

would be a bit better i think, as it:
1. in the standard library
2. exactly what you want.

Reply:
I had been following #4509’s original suggestion that the selection should span newlines, and `char_is_whitespace` returns false for newlines and carriage returns ([https://github.com/helix-editor/helix/blob/master/helix-core/src/chars.rs#L43](https://github.com/helix-editor/helix/blob/master/helix-core/src/chars.rs#L43)). Are you suggesting that it wpould be better to not capture newlines?

Reply:
Never mind, I didn't see that you were suggesting `char::is_whitespace` and not `char_is_whitespace`. I'll apply the changes.